inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_align_node.cpp

```diff
@@ -0,0 +1,371 @@
+
+    if (getParentEdges().size() != 3)
+        THROW_IE_EXCEPTION << errorPrefix << "has incorrect number of input edges: " << getParentEdges().size();
+    if (getChildEdges().empty())
+        THROW_IE_EXCEPTION << errorPrefix << "has incorrect number of output edges: " << getChildEdges().size();
+
+    if (getParentEdgeAt(0)->getDims().ndims() != 4) {
+        THROW_IE_EXCEPTION << errorPrefix << "doesn't support 0th input with rank: " << getParentEdgeAt(0)->getDims().ndims();
+    }
+
+    if (getParentEdgeAt(1)->getDims().ndims() != 2) {
+        THROW_IE_EXCEPTION << errorPrefix << "doesn't support 2st input with rank: " << getParentEdgeAt(2)->getDims().ndims();
+    }
+
+    if (getParentEdgeAt(2)->getDims().ndims() != 1) {
+        THROW_IE_EXCEPTION << errorPrefix << "doesn't support 1st input with rank: " << getParentEdgeAt(1)->getDims().ndims();
+    }
+
+    if (getChildEdgeAt(0)->getDims().ndims() != 4) {
+        THROW_IE_EXCEPTION << errorPrefix << "doesn't support output with rank: " << getChildEdgeAt(0)->getDims().ndims();
+    }
+
+    if (getParentEdgeAt(1)->getDims()[1] != 4) {
+        THROW_IE_EXCEPTION << errorPrefix << "has invalid shape on 1st input: ["
+                           << getParentEdgeAt(1)->getDims()[0] << "," << getParentEdgeAt(1)->getDims()[1] << "]";
+    }
+
+    if (getParentEdgeAt(1)->getDims()[0] != getParentEdgeAt(2)->getDims()[0]) {
+        THROW_IE_EXCEPTION << errorPrefix << "has unequal sizes of proposal inputs for proposals ("
+                           << getParentEdgeAt(1)->getDims()[0] << ") and indexes ("
+                           << getParentEdgeAt(2)->getDims()[0] << ")";
+    }
+
+    pooledH = genericLayer->GetParamAsInt("pooled_h");
+    pooledW = genericLayer->GetParamAsInt("pooled_w");
+    spatialScale = genericLayer->GetParamAsFloat("spatial_scale");
+    samplingRatio = genericLayer->GetParamAsInt("sampling_ratio");
+    std::string m = genericLayer->GetParamAsString("mode", "max");
+    if (m == "max") {
+        opType = ROIAlignOpType::Max;
+    } else if (m == "avg") {
+        opType = ROIAlignOpType::Avg;
+    } else {
+        THROW_IE_EXCEPTION << errorPrefix << "doesn't support roi pooling method: " << m;
+    }
+}
+
+void MKLDNNROIAlignNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    Precision inputPrec0 = getCnnLayer()->insData[0].lock()->getPrecision();
+    Precision inputPrec1 = getCnnLayer()->insData[1].lock()->getPrecision();
+    Precision inputPrec2 = getCnnLayer()->insData[2].lock()->getPrecision();
+    Precision outputPrec = getCnnLayer()->outData[0]->getPrecision();
+
+    if ((inputPrec0 != Precision::FP32 && inputPrec0 != Precision::BF16) ||
+            (inputPrec1 != Precision::FP32) ||
+            (inputPrec2 != Precision::I32) ||
+            (outputPrec != Precision::FP32 && outputPrec != Precision::BF16))
+        return;
+
+    if (!mayiuse(avx512_core_bf16)) {
```

Comment:
Please change to !mayiuse(avx512_core) since the emulator has been added.

Reply:
Done